src/renderers/dom/shared/DOMProperty.js

```diff
@@ -16,6 +16,7 @@ var invariant = require('fbjs/lib/invariant');
 // These attributes should be all lowercase to allow for
 // case insensitive checks
 var RESERVED_PROPS = {
```

Comment:
I realize this isn't part of this change but this seems problematic to me. This is a non-trivial case insensitive look up (as opposed to say a super quick interned string comparison).

We should make reserved props case sensitive and pass other ones through but warn just like we do for any other "possible standard name".